lib/pact_broker/index/service.rb

```diff
@@ -195,13 +195,17 @@ def self.latest_verifications_for_consumer_version_tags(options)
         end
       end
 
-      def self.validate_pacticipant(pacticipant_type, pacticipant_name)
-        error_messages = []
-
-        pacticipant = pacticipant_service.find_pacticipant_by_name(pacticipant_name)
+      def self.filter_item_by_pacticipant_name(index_items, pacticipant_name)
```

Comment:
The query to generate the index results is quite intensive, and can be slow. It's better to filter at the SQL level, not do the entire query and then filter in memory.

Find the matching pacticipants as per the example that I posted earlier, get their IDS, then update the base query to do something like:

```
query = query.where( Sequel.|( { consumer_id: pacticipant_ids}, { provider_id: pacticipant_ids} ) )
```